OnlineStoreRestApiApplication/OnlineStoreRestApiApplication/Controllers/UsersController.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.AspNet.OData;
+using OnlineStoreRestApiApplication.Models;
+using OnlineStoreRestApiApplication.Models.DAL;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web.Http;
+
+namespace UserServices.Controllers
+{
+    public class UsersController : ODataController
+    {
+        UsersDbContext db = new UsersDbContext();
+
+        [EnableQuery]
+        public IQueryable<User> Get()
+        {
+            return db.Users;
+        }
+
+        [EnableQuery]
+        public SingleResult<User> Get([FromODataUri] int key)
+        {
+            IQueryable<User> result = db.Users.Where(p => p.UserID == key);
+            return SingleResult.Create(result);
+        }
+
+        // POST: odata/ItemsOfOrders
+        public async Task<IHttpActionResult> Post(User user)
```

Comment:
You are just taking User ODATA model and saving it to the DB, there should be a couple of more actions:
validate user parameters (such as password, loging and etc. (for ex. for passwored letters, numbers, symbols should be presented and the lenght should be between 8 and 16 symbols)
Then for Password you need to encrypt it somehow (and then when there will be authentification you could decrypt it)